frontend/apps/cloudstorage/webdav.lua

```diff
@@ -17,7 +17,14 @@ function WebDav:run(address, user, pass, path)
 end
 
 function WebDav:downloadFile(item, address, username, password, local_path, callback_close)
-    local code_response = WebDavApi:downloadFile(address .. WebDavApi:urlEncode( item.url ), username, password, local_path)
+    local urlencoded_item_url = WebDavApi:urlEncode( item.url )
+    local download_address = address
+    if string.sub(address, -1 ) == "/" and string.sub(urlencoded_item_url,1,1) == "/" then
```

Comment:
There's no a protocol at this point?

I'll take a look later this week, ping me if I forget. Perhaps this plugin still does some outdated stuff with luasocket

Reply:
Or since you know your way around a bit, see if #6307 applies here.